hibernate-core/src/main/java/org/hibernate/engine/internal/EntityEntryContext.java

```diff
@@ -181,10 +181,13 @@ else if ( nonEnhancedEntityXref == null ) {
 		}
 
 		// if we could not resolve it, just return (it was not associated with this context)
-		if ( managedEntity == null ) {
+		if ( managedEntity == null || managedEntity.$$_hibernate_getEntityEntry() == null) {
```

Comment:
It covers the case where a ManagedEntity has already been evicted using `EntityEntryContext.removeEntityEntry( managedEntity )`.

When `EntityEntryContext.removeEntityEntry( managedEntity )` is called a second time, `managedEntity != null` and `managedEntity.$$_hibernate_getEntityEntry() == null` (because it was already nullified the first time `EntityEntryContext.removeEntityEntry( managedEntity )` was called). `null` should be returned in this case.

As I mentioned, this fix does not take into account HHH-10795. The fix for HHH-10795 will add a new field `EntityEntryContext.persistenceContext`, and an added check will determine if the EntityEntry's PC is the same as the EntityEntryContext's PC. The above check would be changed to something like:
 ```java
if ( managedEntity == null || 
      managedEntity.$$_hibernate_getEntityEntry() == null ||
      persistenceContext != AbstractEntityEntry.class.cast(
            managedEntity.$$_hibernate_getEntityEntry() ).getPersistenceContext() ) {
    return null;
}
```